File: evaluation/S1/java/Easy/390-determine-color-of-a-chessboard-square.java
// Source: https://leetcode.com/problems/determine-color-of-a-chessboard-square/   |   Difficulty: Easy
//
// Problem Description:
// You are given coordinates, a string that represents the coordinates of a square of the chessboard. Below is a chessboard for your reference.
//
//
//
// Return true if the square is white, and false if the square is black.
//
// The coordinate will always represent a valid chessboard square. The coordinate will always have the letter first, and the number second.
//
// Example:
// Input: coordinates = "a1"
// Output: false
// Explanation: From the chessboard above, the square with coordinates "a1" is black, so return false.
//
// Constraints:
// coordinates.length == 2
// 	'a' <= coordinates[0] <= 'h'
// 	'1' <= coordinates[1] <= '8'
//
// Helpful references (internal KB):
// - N/A
//
// Notes:
// - Keep the given method signatures unchanged.
// - Use only Java standard library (`java.*`); no external dependencies.
// - Do not change input/output formats or return value semantics.
// - Conformance: single public class 'Solution'; no 'package' statement; place any extra types as static inner classes.
// - If you introduce any helper functions, include their full implementations in this file; no hidden helpers.

class Solution {
    public boolean squareIsWhite(String coordinates) {
        char file = coordinates.charAt(0); // 'a' to 'h'
        char rank = coordinates.charAt(1); // '1' to '8'

        // Calculate the color based on the sum of file and rank indices
        int fileIndex = file - 'a'; // 0 to 7
        int rankIndex = rank - '1'; // 0 to 7

        // A square is white if the sum of its file and rank indices is odd
        return (fileIndex + rankIndex) % 2 != 0;            
    }
}

File: evaluation/S2/python/Easy/390-determine-color-of-a-chessboard-square.py
# Source: https://leetcode.com/problems/determine-color-of-a-chessboard-square/   |   Difficulty: Easy
#
# Problem Description:
# You are given coordinates, a string that represents the coordinates of a square of the chessboard. Below is a chessboard for your reference.
#
#
#
# Return true if the square is white, and false if the square is black.
#
# The coordinate will always represent a valid chessboard square. The coordinate will always have the letter first, and the number second.
#
# Example:
# Input: coordinates = "a1"
# Output: false
# Explanation: From the chessboard above, the square with coordinates "a1" is black, so return false.
#
# Constraints:
# coordinates.length == 2
# 	'a' <= coordinates[0] <= 'h'
# 	'1' <= coordinates[1] <= '8'
#
# Helpful references (internal KB):
# - N/A
#
# Notes:
# - Keep the given function signature unchanged.
# - Use only the Python standard library; no third-party packages.
# - Do not change input/output formats or return value semantics as defined by the template/problem.
# - Conformance: use features compatible with LeetCode’s Python3 runtime; do NOT rely on undefined/implementation-specific behavior.
# - If you introduce any helper functions/classes, include their full implementations in this file; no hidden helpers.

class Solution(object):
    def squareIsWhite(self, coordinates):
        """
        :type coordinates: str
        :rtype: bool
        """
        # The color of the square can be determined by the sum of the row and column indices.
        # If the sum is even, the square is black; if it's odd, the square is white.
        
        # Convert the letter to a number (a=1, b=2, ..., h=8)
        column = ord(coordinates[0]) - ord('a') + 1
        # Convert the number character to an integer
        row = int(coordinates[1])
        
        # Determine the color based on the parity of the sum of row and column
        return (row + column) % 2 == 1